service/src/main/java/bio/terra/pipelines/stairway/imputation/AddWdsRowStep.java

```diff
@@ -0,0 +1,88 @@
+import bio.terra.pipelines.dependencies.sam.SamService;
+import bio.terra.pipelines.dependencies.stairway.JobMapKeys;
+import bio.terra.pipelines.dependencies.wds.WdsService;
+import bio.terra.pipelines.dependencies.wds.WdsServiceApiException;
+import bio.terra.pipelines.dependencies.wds.WdsServiceException;
+import bio.terra.stairway.*;
+import bio.terra.stairway.exception.RetryException;
+import org.databiosphere.workspacedata.client.ApiException;
+import org.databiosphere.workspacedata.model.RecordAttributes;
+import org.databiosphere.workspacedata.model.RecordRequest;
+
+/**
+ * This step creates or replaces a row to a WDS table specific to the pipeline that was launched
+ * currently it writes the flight id as the primary key and a hardcoded "scatter" value that will be
+ * replaced once inputs are being passed in from the user.
+ *
+ * <p>The step expects a cbas uri to be passed in through the working map
+ */
+public class AddWdsRowStep implements Step {
+  private final WdsService wdsService;
+  private final SamService samService;
+
+  public AddWdsRowStep(WdsService wdsService, SamService samService) {
+    this.wdsService = wdsService;
+    this.samService = samService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    FlightMap inputParameters = flightContext.getInputParameters();
+
+    FlightUtils.validateRequiredEntries(
+        inputParameters,
+        JobMapKeys.PIPELINE_NAME.getKeyName(),
+        RunImputationJobFlightMapKeys.CONTROL_WORKSPACE_ID);
+
+    String controlWorkspaceId =
+        inputParameters.get(RunImputationJobFlightMapKeys.CONTROL_WORKSPACE_ID, String.class);
+    PipelinesEnum pipelineName =
+        inputParameters.get(JobMapKeys.PIPELINE_NAME.getKeyName(), PipelinesEnum.class);
+
+    FlightMap workingMap = flightContext.getWorkingMap();
+    FlightUtils.validateRequiredEntries(workingMap, RunImputationJobFlightMapKeys.WDS_URI);
+
+    String wdsUri = workingMap.get(RunImputationJobFlightMapKeys.WDS_URI, String.class);
+
+    HealthCheckWorkspaceApps.Result healthResult =
+        wdsService.checkHealth(wdsUri, samService.getTspsServiceAccountToken());
+    if (!healthResult.isOk()) {
+      return new StepResult(
+          StepStatus.STEP_RESULT_FAILURE_RETRY,
+          new WdsServiceApiException(new ApiException("WDS is not healthy")));
+    }
+
+    // hardcoded for now until we are using inputs from user
+    RecordAttributes recordAttributes = new RecordAttributes();
+    recordAttributes.put("scatter", 2);
+    RecordRequest createRecordRequest = new RecordRequest().attributes(recordAttributes);
+    try {
+      wdsService.createOrReplaceRecord(
+          wdsUri,
+          samService.getTspsServiceAccountToken(),
+          createRecordRequest,
+          controlWorkspaceId,
+          pipelineName.getValue(),
+          flightContext.getFlightId(),
```

Comment:
it would be helpful to have a comment here noting that we're using the flightId as the primary key in the WDS table

Reply:
I personally think it would be helpful - when I was looking at this I was wondering to myself where/how in this call you're specifying the primary key. you're right that I could have figured it out from the java doc but it wouldn't hurt to have that note here. not a huge deal though, I suspect this may get refactored at some point anyway